htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -568,32 +588,14 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 			// Update detail stock for batch product
 			if (!$error && !empty($conf->productbatch->enabled) && $product->hasbatch() && !$skip_batch)
 			{
-				// check unicity for serial numbered equipments ( different for lots managed products)
-				if ( $product->status_batch == 2 && $qty > 0 )
-				{
-					if ( $this->getBatchCount($fk_product, $batch) > 0 )
```

Comment:
Why removing this test. It is the only test that gurarantee unicity into database.
PS: This test should be moved ate very end of transaction, so at line 642 (after the triggers).

Reply:
@eldy ok for me, latest commit should solve all your points